curvefs/src/mds/topology/topology.cpp

```diff
@@ -1241,27 +1257,26 @@ TopoStatusCode TopologyImpl::ChooseRecoveredMetaServer(
 }
 
 TopoStatusCode TopologyImpl::ChooseAvailableMetaServers(
-    std::set<MetaServerIdType> *metaServers,
-    PoolIdType *poolId) {
+    std::set<MetaServerIdType> *metaServers, PoolIdType *poolId) {
     ReadLockGuard rlockMetaserver(metaServerMutex_);
-    std::vector<const MetaServer*> vec;
+    std::vector<const MetaServer *> vec;
     for (const auto &it : metaServerMap_) {
-        vec.emplace_back(&(it.second));
+        if (it.second.GetOnlineState() == OnlineState::ONLINE) {
+            vec.emplace_back(&(it.second));
+        }
     }
+
     // sort by left disk capacity
-    std::sort(vec.begin(), vec.end(), [](const MetaServer *a,
-        const MetaServer *b) {
-        return static_cast<double>(a->GetMetaServerSpace().GetDiskUsed()) /
-            a->GetMetaServerSpace().GetDiskCapacity() <
-            static_cast<double>(b->GetMetaServerSpace().GetDiskUsed()) /
-            b->GetMetaServerSpace().GetDiskCapacity();});
+    std::sort(vec.begin(), vec.end(),
```

Comment:
Modify the annotation information Line1269.

Reply:
done